contiamo/operational-ui
# Optimize rendering of components

### Summary

Currently, we have a _lot_ of unnecessary rerenders due to improperly implemented event listeners throughout the whole app, _particularly_ in ContextMenu and Select. 

In ContextMenu, the menu itself doesn't even have an `onClick` handler, but listens _on the document_ to toggle a context menu.

Literally just clicking anywhere in the Labs UI currently causes a rerender of the Select and all its children. On _every click_.

![Clicking on the Labs UI](https://user-images.githubusercontent.com/9947422/42938697-77a98df6-8b53-11e8-9ae1-d62652185ef8.gif)


This PR removes some unnecessary components, and removes a _lot_ of unnecessary rerender cycles.

Namely,
- Deleted Checkboxes
- Deleted unnecessary `with-label` higher-order component
- Removed unnecessary renders from:
  - ContextMenu
  - Input
  - Modal
  - Select
  - SidenavHeader
  - TextArea

### Related issue
#605 


### How to Test
Make sure you have the [React DevTools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) in your browser of choice, and then:
- open the inspector
- navigate to the React tab
- enable "Highlight Updates" 
![image](https://user-images.githubusercontent.com/9947422/42938531-e780dac2-8b52-11e8-88ea-d34f8c9be8b1.png)
- click around the demo site and see what's getting rerendered
  - it may be even more optimal to load individual components instead of the whole demo site [like this](https://operational-ui.netlify.com/#!/HeaderBar).


### To be tested

Tester 1

- [ ] No error/warning in the console
- [ ] The netlify build is working
- [ ] Affected components look okay
- [ ] Rerenders are reasonable
  

Tester 2

- [ ] No error/warning in the console
- [ ] The netlify build is working
- [ ] Affected components look okay
- [ ] Rerenders are reasonable
